### PR TITLE
Add MariaDB dump execution and import.

### PR DESCRIPTION
**Pull request by ineilson**
"fetch" script to execute mysqldump to generate a MariaDB dump file (set of SQL statements) from a MariaDB server.
"import" script to execute mysql to import a MariaDB dump file into a MariaDB server.
Both scripts configured using .ini files, including database hosts, usernames and passwords.

### CODE REVIEW

**Comment by gregcorbett** [comment]
Several of the python Codacy issues look like they are easy fixes (the markdown and subprocess ones not so much)

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[flake8]
```
Comment (gregcorbett) [inline .flake8:1]: Can you remove this file?

File: fetchMariaDBdmpFile/failover_fetch.ini_TEMPLATE [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+# Enable to generate an error logfile
+#-log-error=/path/to/failover_backup_error.log
+# Optional verbose logging to log-error
+#-verbose
+
+single-transaction
+routines
+events
+
+# TODO when supported
```
Comment (gregcorbett) [inline fetchMariaDBdmpFile/failover_fetch.ini_TEMPLATE:36]:
```suggestion
# Uncomment below options to enable TLS connections to the database
```

File: importMariaDBdmpFile/failover_import.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    logging.debug('moving ' + importPath + ' to ' + archivePath)
+
+    shutil.move(importPath, archivePath)
+
+    logging.debug('removing all .dmp files in ' + archive)
+
+    for oldDump in glob.iglob(archive + '/*.dmp'):
+        os.remove(oldDump)
+
+    shutil.move(archivePath, archivePath + '.dmp')
```
Comment (gregcorbett) [inline importMariaDBdmpFile/failover_import.py:203]: Why not delete all older `.dmp` files then do a single `shutil.move(importPath, archivePath + '.dmp')`?

File: .flake8 (hunk 1 above)
Comment (ineilson) [reply .flake8:1]: Removed.

File: fetchMariaDBdmpFile/failover_fetch.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        if os.path.isfile(cnf.noFetch):
+            logging.error(cnf.noFetch + ' exists. No fetch attempted. File contents -')
+            with open(cnf.noFetch) as f:
+                logging.error(f.read().rstrip())
+            return 1
+
+        runDump(cnf.mysqlOptionsPath, cnf.databaseName)
+
+        archiveDump(cnf.resultFile, cnf.dumpFile)
+
```
[the remark this answers is not in the document]
Comment (ineilson) [reply fetchMariaDBdmpFile/failover_fetch.py:132]: The dump is 'attempted' to a temporary/staging file. The archive will only happen if the dump is successful. This mirrors the logic of the existing Oracle flow - or is supposed to.

File: importMariaDBdmpFile/failover_import.py (hunk 3 above)
Comment (ineilson) [reply importMariaDBdmpFile/failover_import.py:203]: I can't think of a good reason, except for slavishly following existing logic. I'll rework it.

File: importMariaDBdmpFile/failover_import.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+    logging.debug('removing all .dmp files in ' + archive)
+
+    for oldDump in glob.iglob(archive + '/*.dmp'):
+        os.remove(oldDump)
+
+    archivePath = archive + '/' + strftime(format, gmtime()) + '.dmp'
+
+    logging.debug('moving ' + importPath + ' to ' + archivePath)
+
+    shutil.move(importPath, archivePath)
```
Comment (gregcorbett) [inline importMariaDBdmpFile/failover_import.py:201]:
```suggestion
    shutil.move(importPath, archivePath + '.dmp')
```

Comment (ineilson) [reply importMariaDBdmpFile/failover_import.py:201]: Appended '.dmp' in the archivePath assignment for use in logging.debug stmt.

**Review by gregcorbett — APPROVED** [review]
LGTM

**Pull request merged**